### ISSUE TITLE
Tuner function not accessible in docker

### ISSUE THREAD

**Issue by gitchat1** [remsky/Kokoro-FastAPI#537]
Hi, I wanted to test the new runer fuction. So I git pulled the latest files and then ran docker compose up --build. I can see the tuner option in the webui it is not clickable though here is the log.

(base) anonym@anonym-Victus-by-HP-Gaming-Laptop-15-fb2xxx:~/Kokoro-FastAPI/docker/gpu$ sudo docker compose up
Attaching to kokoro-tts-1
kokoro-tts-1  | Model files already exist and are valid
kokoro-tts-1  | ✓ Inno tuner weights prepared at /app/api/src/models/v1_0/inno_tuner/model.safetensors
kokoro-tts-1  | Global API loguru logger level: DEBUG
kokoro-tts-1  | INFO:     Started server process [1]
kokoro-tts-1  | INFO:     Waiting for application startup.
kokoro-tts-1  | 01:30:31 PM | INFO     | main:68 | Loading TTS model and voice packs...
kokoro-tts-1  | 01:30:31 PM | INFO     | model_manager:46 | Initializing Kokoro V1 on cuda
kokoro-tts-1  | 01:30:31 PM | DEBUG    | paths:131 | Searching for model in path: /app/api/src/models
kokoro-tts-1  | 01:30:31 PM | INFO     | kokoro_v1:162 | Loading Kokoro model on cuda
kokoro-tts-1  | 01:30:31 PM | INFO     | kokoro_v1:163 | Config path: /app/api/src/models/v1_0/config.json
kokoro-tts-1  | 01:30:31 PM | INFO     | kokoro_v1:164 | Model path: /app/api/src/models/v1_0/kokoro-v1_0.pth
kokoro-tts-1  | /app/.venv/lib/python3.12/site-packages/torch/nn/modules/rnn.py:123: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.2 and num_layers=1
kokoro-tts-1  |   warnings.warn(
kokoro-tts-1  | /app/.venv/lib/python3.12/site-packages/torch/nn/utils/weight_norm.py:144: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
kokoro-tts-1  |   WeightNorm.apply(module, name, dim)
kokoro-tts-1  | 01:30:32 PM | DEBUG    | paths:171 | Scanning for voices in path: /app/api/src/voices/v1_0
kokoro-tts-1  | 01:30:32 PM | DEBUG    | paths:155 | Searching for voice in path: /app/api/src/voices/v1_0
kokoro-tts-1  | 01:30:32 PM | DEBUG    | model_manager:85 | Using default voice 'af_heart' for warmup
kokoro-tts-1  | 01:30:32 PM | DEBUG    | kokoro_v1:130 | Cached voice tensor from /app/api/src/voices/v1_0/af_heart.pt
kokoro-tts-1  | 01:30:32 PM | INFO     | kokoro_v1:201 | Creating new pipeline for language code: a
kokoro-tts-1  | 01:30:32 PM | DEBUG    | kokoro_v1:382 | Generating audio for text with lang_code 'a': 'Warmup text for initialization.'
kokoro-tts-1  | 01:30:33 PM | DEBUG    | kokoro_v1:389 | Got audio chunk with shape: torch.Size([57600])
kokoro-tts-1  | 01:30:33 PM | INFO     | model_manager:92 | Warmup completed in 1712ms
kokoro-tts-1  | 01:30:33 PM | INFO     | main:129 | 
kokoro-tts-1  | 
kokoro-tts-1  | ░░░░░░░░░░░░░░░░░░░░░░░░
kokoro-tts-1  | 
kokoro-tts-1  |     ╔═╗┌─┐┌─┐┌┬┐
kokoro-tts-1  |     ╠╣ ├─┤└─┐ │ 
kokoro-tts-1  |     ╚  ┴ ┴└─┘ ┴
kokoro-tts-1  |     ╦╔═┌─┐┬┌─┌─┐
kokoro-tts-1  |     ╠╩╗│ │├┴┐│ │
kokoro-tts-1  |     ╩ ╩└─┘┴ ┴└─┘
kokoro-tts-1  | 
kokoro-tts-1  | ░░░░░░░░░░░░░░░░░░░░░░░░
kokoro-tts-1  |                 
kokoro-tts-1  | Model warmed up on cuda: kokoro_v1
kokoro-tts-1  | CUDA: True
kokoro-tts-1  | 72 voice packs loaded
kokoro-tts-1  | 
kokoro-tts-1  | Beta Web Player: http://0.0.0.0:8880/web/
kokoro-tts-1  | or http://localhost:8880/web/
kokoro-tts-1  | ░░░░░░░░░░░░░░░░░░░░░░░░
kokoro-tts-1  | 
kokoro-tts-1  | INFO:     Application startup complete.
kokoro-tts-1  | INFO:     Uvicorn running on http://0.0.0.0:8880 (Press CTRL+C to quit)

**Comment by remsky** [comment]
Hey @gitchat1 

The endpoint is disabled by default currently, but if you uncomment the `ENABLE_INNO_TUNER=true` in the docker compose, you can re-run 

```bash
docker compose up
```

and it should be enabled for you without needing a rebuild. 

I'll add that in as a tooltip on the WebUI, but let me know if you still have any issue. [docs/inno-tune.md](https://github.com/remsky/Kokoro-FastAPI/blob/master/docs/inno-tune.md) also has extra details

**Comment by gitchat1** [comment]
Hi, so that did exactly nothing. Here's what my docker file looked after uncommenting ENABLE_INNO_TUNER=true.

name: kokoro-fastapi-gpu
services:
  kokoro-tts:
    image: ghcr.io/remsky/kokoro-fastapi-gpu:${VERSION:-latest}
    build:
      context: ../..
      dockerfile: docker/gpu/Dockerfile.optimized
      cache_from:
        - ghcr.io/remsky/kokoro-fastapi-cache:gpu-amd64
        - ghcr.io/remsky/kokoro-fastapi-cache:gpu-arm64
      # Default builds cu126 wheels, multi-arch (Maxwell through Ada: GTX 9xx, 10xx, 20xx, 30xx, 40xx).
      # Temp fix for NVIDIA Blackwell / RTX 50-series, uncomment:
      # args:
      #   GPU_EXTRA: gpu-cu128
      #   CUDA_VERSION: 12.8.1
    volumes:
      - ../../web:/app/web  # hot reload changes to web player code
    ports:
      - "8880:8880"
    env_file:
      - path: ../../.env
        required: false
    environment:  # full list in docs/configuration.md
      - API_LOG_LEVEL=DEBUG
       - ENABLE_INNO_TUNER=true
      # - ALLOW_DEV_UNLOAD=true
      # - ENABLE_DEBUG_ENDPOINTS=true
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: all
              capabilities: [gpu]

____________________

I also tried to rebuilt with the zbcomented file there was no change. I then tried uncommenting the   # - ALLOW_DEV_UNLOAD=true
      # - ENABLE_DEBUG_ENDPOINTS=true parameters but that didn't change anything either.
I should mention that I was using conda at the time although thzat shouldn't have any impact on docker. I also installed the UV version of the programme just to see if the tune fuctionality was available there. None of that should have influenced the docker installatzion as far as I know.

**Comment by remsky** [comment]
Can you paste the startup banner from the new run? I tested on my end, and it is able to toggle on/off. With the flag on there's an extra line under the voice pack count:

```
...
* voice packs loaded
Inno voice tuner: ready at /dev/tune
...
```

If that line isn't there, the env var isn't reaching the container at all; I'd try docker compose down then docker compose up to force a recreate. If you're running a uv version as well, I'd also double check that it is fully shut down/killed (not shadowing the port somehow). 

If none of the above resolves, can you try a direct docker run command on a fresh port and check ` localhost:8881/web/`  e.g.

```bash
docker run --gpus all -e ENABLE_INNO_TUNER=true -p 8881:8880 ghcr.io/remsky/kokoro-fastapi-gpu:latest
```

**Comment by gitchat1** [comment]
Okay this is very weird No matter what I do with the docker compose file that line Inno voice tuner: ready at /dev/tune is always missing. When I use docker run --gpus all -e ENABLE_INNO_TUNER=true -p 8881:8880 ghcr.io/remsky/kokoro-fastapi-gpu:latest it is there but no matter what I do I can't open the webinterface at all. I just see the screen you get when a website is unavailable. This is the log.
______________

sudo docker run --gpus all -e ENABLE_INNO_TUNER=true -p 8881:8880 ghcr.io/remsky/kokoro-fastapi-gpu:latest Model files already exist and are valid ✓ Inno tuner weights prepared at /app/api/src/models/v1_0/inno_tuner/model.safetensors Global API loguru logger level: DEBUG INFO:     Started server process [1] INFO:     Waiting for application startup. 04:49:38 PM | INFO     | main:68 | Loading TTS model and voice packs... 04:49:38 PM | INFO     | model_manager:46 | Initializing Kokoro V1 on cuda 04:49:38 PM | DEBUG    | paths:131 | Searching for model in path: /app/api/src/models 04:49:38 PM | INFO     | kokoro_v1:162 | Loading Kokoro model on cuda 04:49:38 PM | INFO     | kokoro_v1:163 | Config path: /app/api/src/models/v1_0/config.json 04:49:38 PM | INFO     | kokoro_v1:164 | Model path: /app/api/src/models/v1_0/kokoro-v1_0.pth /app/.venv/lib/python3.12/site-packages/torch/nn/modules/rnn.py:123: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.2 and num_layers=1 warnings.warn( /app/.venv/lib/python3.12/site-packages/torch/nn/utils/weight_norm.py:144: FutureWarning: torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm. WeightNorm.apply(module, name, dim) 04:49:39 PM | DEBUG    | paths:171 | Scanning for voices in path: /app/api/src/voices/v1_0 04:49:39 PM | DEBUG    | paths:155 | Searching for voice in path: /app/api/src/voices/v1_0 04:49:39 PM | DEBUG    | model_manager:85 | Using default voice 'af_heart' for warmup 04:49:39 PM | DEBUG    | kokoro_v1:130 | Cached voice tensor from /app/api/src/voices/v1_0/af_heart.pt 04:49:39 PM | INFO     | kokoro_v1:201 | Creating new pipeline for language code: a 04:49:40 PM | DEBUG    | kokoro_v1:382 | Generating audio for text with lang_code 'a': 'Warmup text for initialization.' 04:49:40 PM | DEBUG    | kokoro_v1:389 | Got audio chunk with shape: torch.Size([57600]) 04:49:40 PM | INFO     | model_manager:92 | Warmup completed in 2088ms 04:49:41 PM | INFO     | inno_tuner:40 | Inno voice tuner v0.2.0 loaded on cuda 04:49:41 PM | INFO     | main:129 |
░░░░░░░░░░░░░░░░░░░░░░░░
╔═╗┌─┐┌─┐┌┬┐
╠╣ ├─┤└─┐ │ 
╚  ┴ ┴└─┘ ┴
╦╔═┌─┐┬┌─┌─┐
╠╩╗│ │├┴┐│ │
╩ ╩└─┘┴ ┴└─┘
░░░░░░░░░░░░░░░░░░░░░░░░
Model warmed up on cuda: kokoro_v1 CUDA: True 72 voice packs loaded Inno voice tuner: ready at /dev/tune
Beta Web Player: http://0.0.0.0:8880/web/ or http://localhost:8880/web/ ░░░░░░░░░░░░░░░░░░░░░░░░
INFO:     Application startup complete. INFO:     Uvicorn running on http://0.0.0.0:8880 (Press CTRL+C to quit)

_________

I also feel obligated to point out that I've never been able to actually build the docker project on a clean system using your docker file that always errored out from the first version to the latest. I always had to use the following file on initial setup. 

name: kokoro-tts-gpu
services:
  kokoro-tts:
    build:
      context: ../..
      dockerfile: docker/gpu/Dockerfile.optimized
    volumes:
      - api_volume:/app/api
    ports:
      - "8880:8880"
    environment:
      - PYTHONPATH=/app
      - USE_GPU=true
      - PYTHONUNBUFFERED=1
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: all
              capabilities: [gpu]

volumes:
  api_volume:
    name: kokoro-tts-api

_________

Once initial setup was done however I've never experienced any problem updating the project until now, even when using your provided file.
As for the uv version I don't exactly know how to make sure nothings running in the background but I presume restarting the system and then only starting the docker version should be good enough. Is there a way to permanently remove the uv version so it can't interfere?

**Comment by remsky** [comment]
It's hard to tell if it's a copy-paste bug through github, but the raw values of your compose file in the earlier comment also show an additional indent/space prefixing the line of the `ENABLE_INNO_TUNER=true` which would explain it not being picked up. 

- Can you double-check/retype that line, just to confirm it's aligned with the other environment lines. 

And sorry, to make sure I'm following. If that doesn't resolve it:

- What was the error you were seeing on the docker compose that you had to work around? 
- When you run that docker run command for the alternate, are you checking localhost:**_8881_**/web/ for the UI? 
- If the systems restarted, the uv isn't it, so all good there. 

Can you also share the output of these commands? And details on how/where you're running this? (operating system, etc):

```bash
docker version --format '{{.Server.Version}}'
docker compose version
docker buildx version
echo "DOCKER_BUILDKIT=${DOCKER_BUILDKIT:-unset}"
```

If it helps, it also shouldn't require a whole build unless you're making custom changes, something like this be a simpler base to compose up from, and should run from any location a fair bit quicker. 

```yaml
name: kokoro-tts-gpu
services:
  kokoro-tts:
    image: ghcr.io/remsky/kokoro-fastapi-gpu:latest
    ports:
      - "8880:8880"
    environment:
      - API_LOG_LEVEL=DEBUG
      - ENABLE_INNO_TUNER=true
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: all
              capabilities: [gpu]

```

**Comment by raf-alencar** [comment]
I just tried to see if I could replicate the issue, and didn't see it. it's working fine on my machine.

**Comment by gitchat1** [comment]
I just deleted everything and reinstalled from scratch. Now It works perfectly fine. Can you tell me how to uninstall the UV version so I can save some space?

**Comment by remsky** [comment]
@gitchat1 Great, and glad to hear it.

If you delete the .`venv` in the folder (and `api/src/models/v1_0` for the model), then run `uv cache clean`, that should clear the space for you.
